### PR TITLE
PG: ignore info from down osd

### PR DESCRIPTION
**Pull request by athanatos**
Fixes: #12990
Backport: firefly, hammer
Signed-off-by: Samuel Just <sjust@redhat.com>

### CODE REVIEW

**Comment by athanatos** [comment]
Tested, ready for merge.

File: src/osd/PG.cc [hunk 1]
```diff
@@ -324,6 +324,12 @@ bool PG::proc_replica_info(pg_shard_t from, const pg_info_t &oinfo)
     return false;
   }
 
+  if (get_osdmap()->is_down(from.osd)) {
```
Comment (liewegas) [inline src/osd/PG.cc]: shouldn't this actually check is_down(from.osd) || get_up_from(from.osd) > sender's map_epoch?

**Comment by athanatos** [comment]
Updating, will retest

**Comment by athanatos** [comment]
Closing to retarget at infernalis

**Pull request closed without merging**